tests/distributions/test_multivariate.py

```diff
@@ -1634,6 +1634,12 @@ def logp_norm(value, sigma, axes):
 
         np.testing.assert_allclose(zsn_logp, mvn_logp)
 
+    def test_does_not_upcast_to_float64(self):
+        with pytensor.config.change_flags(floatX="float32", warn_float64="raise"):
+            with pm.Model():
+                pm.ZeroSumNormal("b", sigma=1, shape=(2,))
+                pm.sample(1, chains=1, tune=1)
```

Comment:
Should be enough to call `model.logp()`

Reply:
I do not think it is enough because `logp` does not call `ZeroSumTransform.backward`. `pm.sample` is a way to test both `ZeroSumTransform.backward` and `logp`. I added a comment about this interaction in [`7906292` (#6889)](https://github.com/pymc-devs/pymc/pull/6889/commits/79062922e2ce10865b03d6473cd202cef8f484c5)

Note that the new test is mostly a non-regression test for https://github.com/pymc-devs/pymc/issues/6886.

Reply:
Ah I see how `m.logp` works. Thanks for the suggestion.